tests/ui/return/return-from-residual-sugg-issue-125997.stderr

```diff
@@ -0,0 +1,39 @@
+error[E0277]: the `?` operator can only be used in a function that returns `Result` or `Option` (or another type that implements `FromResidual`)
+  --> $DIR/return-from-residual-sugg-issue-125997.rs:10:44
+   |
+LL | fn test1() {
+   | ---------- this function should return `Result` or `Option` to accept `?`
+LL |     let mut _file = File::create("foo.txt")?;
+   |                                            ^ cannot use the `?` operator in a function that returns `()`
+   |
+   = help: the trait `FromResidual<Result<Infallible, std::io::Error>>` is not implemented for `()`
+help: consider adding return type
+   |
+LL ~ fn test1() -> Result<(), Box<dyn std::error::Error>> {
+LL ~     let mut _file = File::create("foo.txt")?;
```

Comment:
why does this line appear as modified?

Reply:
> why does this line appear as modified?

I think this is because the span I used when suggesting `return Ok(());` is not quite precise.
I use `s.span.shrink_to_hi().until(body.value.span.shrink_to_hi()),`  to give return suggestion, `s` is the end Stmt of the block, in this code, `s` is `let mut _file = File::create("foo.txt")?;`

Reply:
Is there any better way here? Thank you.

Reply:
Done. Thank you.